app/src/main/kotlin/com/ayatk/biblio/util/ext/IntentExt.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016-2018 ayatk.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ayatk.biblio.util.ext
+
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import java.io.Serializable
+
+fun Intent.extraOf(vararg pairs: Pair<String, Any>) = this.apply {
```

Comment:
**[detekt]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Prefer splitting up complex methods into smaller, easier to understand methods.